### ISSUE TITLE
Throw an error

### ISSUE THREAD

**Issue by mjackson** [alexreardon/tiny-warning#2]
One thing that I always really liked about FB's warning package is that they [`throw` an error](https://github.com/BerkeleyTrue/warning/blob/e4806228c076429774878522cafdf9321b7ff71c/warning.js#L36-L41) to let the DevTools hook into the stack trace so you can find the code that triggered the warning.

Any interest in adding that feature here in dev mode?

**Comment by alexreardon** [comment]
Sorry I missed this @mjackson 

While that might be useful for devtools, I think it breaks the mental modal of what a 'warning' is. Throwing an error is a different control flow.

How I see it:

- `warning`: you are not doing something correctly and you might have issues - but we will press on
- `invariant`: something bad has happened and we are going to throw an error to kill the existing flow

**Comment by mjackson** [comment]
They don't actually *throw* like usual. They `throw` and then immediately `catch` the error so it doesn't leak out into your code. The point is to generate a stack trace and let you look into the call site using the browser's devtools.

**Comment by alexreardon** [comment]
This one?

https://github.com/BerkeleyTrue/warning/blob/master/warning.js#L36

```js
try {
      // --- Welcome to debugging React ---
      // This error was thrown as a convenience so that you can use this stack
      // to find the callsite that caused this warning to fire.
      throw new Error(message);
    } catch (x) {}
```

Sounds useful! PR welcome 😊

**Comment by alexreardon** [comment]
Closed by #6
